File: rosbag_manager/scripts/bag_clock_counter.py
#! /usr/bin/env python
import rospy
from rosgraph_msgs.msg import Clock

class BagClockCounter:
    def __init__(self, hz = 100):
        self.count = 0
        self.fase = 0
        self.switch_flag = False
    def __del__(self):
        print("!!!!!!!!!!!!!!1")
    def counterCallback(self, msg):
        self.count += 1
    def need_switch_fase(self):
        #print("fase:{}".format(self.fase))
        #print("count:{}".format(self.count))
        if(self.fase == self.count):
            self.switch_flag = False
            return False
        else:
            self.fase = self.count
            self.switch_flag = True

            return True

File: rosbag_manager/src/bag_clock_counter.cpp
#include "bag_clock_counter.h"

BagClockCounter::BagClockCounter(int hz_):hz(hz_){

    t = ros::Time::now();
    count = 0;
    fase = 0;
    duration = 0.0;
}
BagClockCounter::~BagClockCounter(){


}
void BagClockCounter::counterCallback(const rosgraph_msgs::Clock& msg){
    count++;
    std::cout << "*****time*****: " <<  msg.clock << std::endl;
}
bool BagClockCounter::need_switch_fase(){
    if(fase == count ){
        return false;
    }
    else{
        fase = count;
        std::cout <<"fase: "<< fase << std::endl;
        return true;
    }
}
